Various fixes to C# documentation generator

raulsntos opened the pull request:
- Fix references to global constants and members of `@GlobalScope`
- Log errors for missing members or methods
- Finds referenced members that are derived (they were not found before since it was only looking in the current type)
- Hardcodes special case for `_init` method, in C# replaced with a reference to the constructor
- Ignores properties with slashes (since they are not declared in C# and can't be referenced)
- Add support for `theme_item` (they are not declared in C# and can't be referenced, but before they were not even included in the documentation now they are added with the `<c>` tag)
- Refactor `bbcode_to_xml` method

raulsntos committed:
Fix references to global constants in C# documentation generator

Tries to find the referenced constants in the GlobalScope
if not found in the target class or if no class is specified.

raulsntos committed:
Check for missing methods/members in C# documentation generator

- Outputs errors for missing members or methods when generating the C#
documentation.
- Hardcodes a special case for the `_init` method, in C# we'll reference
the constructor.
- Ignores properties with slashes (since they are not declared in C# and
can't be referenced in the documentation).

raulsntos committed:
Fix checking for `@GlobalScope` in C# documentation generator

raulsntos committed:
Add theme_item support to C# documentation generator

raulsntos committed:
Find inherited members in C# documentation generator

neikeq commented on modules/mono/editor/bindings_generator.cpp:
```diff
@@ -414,20 +457,8 @@ String BindingsGenerator::bbcode_to_xml(const String &p_bbcode, const TypeInterf
-					xml_output.append("</c>");
-				} else if (!target_itype && target_cname == name_cache.type_at_GlobalScope) {
+				if (link_target_parts[0] == name_cache.type_at_GlobalScope) {
+				find_constant_in_global_scope:
```
Nitpicking: I would prefer `find_constant_in_global_scope` to be a function.

raulsntos replied:
I've extracted the BBCode tags XML generation to their own separate method. It's a separate commit so let me know if you don't like it and I can just drop it or redo it differently.

raulsntos committed:
Refactor `bbcode_to_xml` into multiple specific methods

The specific `_append_xml_*` methods implement the logic that generates
the proper XML documentation for the given BBCode tag and target and
appends it to the output.

akien-mga commented:
Thanks!
